lib/reversetunnel/srv.go

```diff
@@ -173,35 +173,31 @@ func (s *server) Close() error {
 }
 
 func (s *server) HandleNewChan(conn net.Conn, sconn *ssh.ServerConn, nch ssh.NewChannel) {
-	log.Infof("got new channel request: %v", nch.ChannelType())
-	switch nch.ChannelType() {
-	case chanHeartbeat:
-		log.Infof("got heartbeat request from agent: %v", sconn)
-		var site *tunnelSite
-		var remoteConn *remoteConn
-		var err error
-
-		switch sconn.Permissions.Extensions[extCertType] {
-		case extCertTypeHost:
-			site, remoteConn, err = s.upsertSite(conn, sconn)
-		default:
-			err = trace.BadParameter("can't retrieve certificate type in certType")
-		}
-
-		if err != nil {
-			log.Errorf("failed to upsert site: %v", err)
-			nch.Reject(ssh.ConnectionFailed, "failed to upsert site")
-			return
-		}
-
-		ch, req, err := nch.Accept()
-		if err != nil {
-			log.Errorf("failed to accept channel: %v", err)
-			sconn.Close()
-			return
-		}
-		go site.handleHeartbeat(remoteConn, ch, req)
+	if nch.ChannelType() != chanHeartbeat {
+		log.Warningf("reversetunnel received unknown channel request %v from %v",
```

Comment:
I think here you'd better do `nch.Reject` to tell client the error and not just hang up